### ISSUE TITLE
Disambiguate "byte" size

### ISSUE THREAD

**Issue by Rudxain** [rust-lang/reference#1673]
[The Glossary](https://github.com/rust-lang/reference/blob/5110ba486b7ad8750f8f9c26000189416a1aca75/src/glossary.md?plain=1#L207-L220) and ["Type Layout"](https://github.com/rust-lang/reference/blob/5110ba486b7ad8750f8f9c26000189416a1aca75/src/type-layout.md) don't specify the size of a "byte".

- Was that intentional?
- Is it a real byte? (platform-dependent smallest unit of addressable memory?)
- Is it guaranteed to be an octet?

I think it's worth being explicit about that.

BTW, the [`bool` reference](https://github.com/rust-lang/reference/blob/5110ba486b7ad8750f8f9c26000189416a1aca75/src/types/boolean.md?plain=1#L20-L26) seems somewhat contradictory:
- `size_of::<bool>() == 1` implies platform-dependent byte
- The hexadecimal bit-patterns imply `u8` (guaranteed octet)

> [!note]
> Previous discussions:
> - rust-lang/rust#46176
> - rust-lang/rust#46156

IMO, bytes are more honest and correct than octets (for `bool` and some `enum`s, at least). For primitives such as `char` and `u8`, and exact bit-count is better than bytes

**Comment by bjorn3** [comment]
I don't think we support any target where a byte is anything other than 8 bits, just like we don't support any systems that use something other than two's complement for signed integers.
